src/components/Popup.js

```diff
@@ -1,39 +1,41 @@
 class Popup {
-    constructor ({popupSelector}) {
-        this._popupElement = document.querySelector(popupSelector);
-        this._handleEscClose = this._handleEscClose.bind(this);
-        this._closeButton = this._popupElement.querySelector('.modal__close');
-        this._addEventListeners();
-    }
-
-    _addEventListeners() {
-        this._closeButton.addEventListener('click', this.close.bind(this));
-    }
+  constructor({ popupSelector }) {
+    this._popupElement = document.querySelector(popupSelector);
+    this._handleEscClose = this._handleEscClose.bind(this);
+    this._closeButton = this._popupElement.querySelector(".modal__close");
+    //this._addEventListeners();
+  }
 
-    open() {
-       this._popupElement.classList.add('modal_opened');
-       document.addEventListener('keydown', this._handleEscClose);
-    }
+  //_addEventListeners() {
+  //}
 
-    close() {
-        this._popupElement.classList.remove('modal_opened');
-        document.removeEventListener('keydown', this._handleEscClose);
-    }
+  open() {
+    this._popupElement.classList.add("modal_opened");
+    document.addEventListener("keydown", this._handleEscClose);
+  }
 
-    _handleEscClose(evt) {
+  close() {
+    this._popupElement.classList.remove("modal_opened");
+    document.removeEventListener("keydown", this._handleEscClose);
+  }
 
-        if (evt.key === "Escape") {
-            this.close();
-        }
+  _handleEscClose(evt) {
+    if (evt.key === "Escape") {
+      this.close();
     }
+  }
 
-    setEventListeners() {
-        this._popupElement.addEventListener("click", (evt) => {
-            if (evt.target.classList.contains("modal") || evt.target.classList.contains('.modal__close')) {
-                this.close();
-            }
-        });
-    }
+  setEventListeners() {
+    this._popupElement.addEventListener("click", (evt) => {
+      if (
+        evt.target.classList.contains("modal") ||
+        evt.target.classList.contains(".modal__close")
```

Comment:
`        evt.target.classList.contains(".modal__close")
`

This doesn't work. Please, delete it


You handle the close icon in line 37, but here will be only the overlay handling

```js

      if (evt.target.classList.contains("modal")) {
        this.close();
      }
```